src/ts/deploy.ts

```diff
@@ -1,4 +1,8 @@
 import { utils } from "ethers";
+import { Artifact } from "hardhat/types";
+
+import authenticatorArtifacts from "../../build/artifacts/src/contracts/GPv2AllowListAuthentication.sol/GPv2AllowListAuthentication.json";
+import settlementArtifacts from "../../build/artifacts/src/contracts/GPv2Settlement.sol/GPv2Settlement.json";
```

Comment:
Something weird happening here.... Are we sure `*.sol/*.json` is a valid import path?

<img width="906" alt="Screenshot 2020-12-14 at 15 16 01" src="https://user-images.githubusercontent.com/11778116/102091618-5e2f5280-3e1f-11eb-91dc-c47897cb3d9a.png">

Reply:
I changed the compilation order and added details to the description of this PR.